### PR TITLE
SBF additions

### PR DESCRIPTION
**Pull request by SJuliez**
This adds bits and pieces to SBF gameplay with comparatively minor changes to TW-related or general classes.

This is probably extremely tedious to review. I suggest looking at the general client/server classes only and ignore everything called SBF... 

When this is merged, we could move some TW-related classes to their own package (GameManager).

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/tooltip/SBFFormationTooltip.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        String styleColor = Integer.toHexString(ownerColor.getRGB() & 0xFFFFFF);
+        result.append("<div style=\"padding:0 10; border:2; border-style:solid; border-color:" + styleColor + ";\">");
+        result.append(nameLines(formation, game));
+        result.append(formationStats(formation));
+        result.append("</div>");
+        result.append("</BODY></HTML>");
+        return result.toString();
+    }
+
+    private static String asCSS(String cssClass, String content) {
```
Comment (kuronekochomusuke) [inline megamek/src/megamek/client/ui/swing/tooltip/SBFFormationTooltip.java:71]: should these be an a common html utils class?

File: megamek/src/megamek/client/ui/swing/tooltip/SBFFormationTooltip.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        return "<span class=\"" + cssClass + "\">" + content + "</span>";
+    }
+
+    private static StringBuilder styles() {
+        float base = UIUtil.scaleForGUI(UIUtil.FONT_SCALE1);
+        int labelSize = (int) (0.8 * base);
+        int valueSize = (int) (1.1 * base);
+        int nameSize = (int) (1.3 * base);
+
+        StringBuilder result = new StringBuilder("<style>");
```
Comment (kuronekochomusuke) [inline megamek/src/megamek/client/ui/swing/tooltip/SBFFormationTooltip.java:93]: is the </style> missing?

File: megamek/src/megamek/client/ui/swing/tooltip/SBFFormationTooltip.java (hunk 1 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/tooltip/SBFFormationTooltip.java:71]: Actually that class was no longer used, but true. Will change.

File: megamek/src/megamek/client/ui/swing/tooltip/SBFFormationTooltip.java (hunk 2 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/tooltip/SBFFormationTooltip.java:93]: Actually that class was no longer used, but true. Will change.

File: megamek/src/megamek/client/ui/swing/tooltip/SBFFormationTooltip.java (hunk 1 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/tooltip/SBFFormationTooltip.java:71]: Yes possibly. Will have a look

**Review by kuronekochomusuke — APPROVED** [review]

**Comment by IllianiCBT** [comment]
image
Seeing this got me unreasonably excited. I'll get this merged prior to tonights' build, assuming Hammer doesn't get there first.

**Pull request merged**